AutoPBW/Game.cs

```diff
@@ -346,18 +351,35 @@ public void DownloadTurn()
 		public void UploadEmpire(string empfile)
 		{
 			var url = "http://pbw.spaceempires.net/games/{0}/player-empire/upload".F(Code);
-			var path = Path.Combine(Path.GetDirectoryName(Engine.PlayerExecutable).Trim('"'), Mod.EmpirePath);
+			var path = Path.Combine(Engine.GetPlayerExecutableDirectory(), Mod.EmpirePath);
 			PBW.Log.Write($"Uploading empire {path} for {this}.");
 			ArchiveUploadAndDeleteArchive(new string[] { empfile }, url, "emp_file");
 		}
 
+		/// <summary>
+		/// Checks if we have a turn file pending upload
+		/// </summary>
+		public bool IsReadyToUploadTurn()
+		{
+			if (Status == PlayerStatus.Waiting)
+			{
+				// get list of files
+				var path = GetSavePath();
+				var files = GetFiles(path, GenerateArgumentsOrFilter(Engine.PlayerTurnUploadFilter));

```

Comment:
Don't we need to make sure that the PLR file has a later timestamp than the GAM file? After all, SE4 and SE5 reuse PLR file names from turn to turn. Problem is, PBW file timestamps seem to operate on UTC, so I'm not sure how we would compare that to a local timestamp on the PLR file. `DateTimeOffset` maybe? Alternately, we could delete the PLR file when uploading it to PBW, or archive it in a folder, or set a flag somewhere, to make sure we don't upload the same file twice.

Reply:
You're right, we need to do that. Shouldn't be too hard to do I think.

Reply:
The turn start date (when the gam was uploaded, in UTC) is sent as part of the xml, so we can compare the plr file timestamp against that.